### DISCUSSION TITLE
Only works right in debug mode

### DISCUSSION THREAD

**Discussion by JamesOsborn-SE** [ImageMagick/ImageMagick#7202] [Help]
### ImageMagick version

7.1.1-29 Q16-HDRI x86_64 21991

### Operating system, version and so on

Endevor Linux 6.8.1-arch1-1 #1 SMP PREEMPT_DYNAMIC

### Description

only works with debug

```shell

# doesn't work as expected, output is L.png only
composite L.png D.png rect_mask.png oof.png

# works as expected
composite -debug all L.png D.png rect_mask.png oof-debug.png
```

```shell
# magick identify +ping L.png 
L.png PNG 1080x2209 1080x2209+0+0 8-bit TrueColorAlpha sRGB 93439B 0.220u 0:00.056

# magick identify +ping D.png
D.png PNG 1080x2209 1080x2209+0+0 8-bit TrueColorAlpha sRGB 93001B 0.220u 0:00.052

# magick identify +ping rect_mask.png
rect_mask.png PNG 1080x2209 1080x2209+0+0 16-bit GrayscaleAlpha Gray 34201B 0.210u 0:00.046
```
files:
[imagemagickTest.zip](https://github.com/ImageMagick/ImageMagick/files/14799405/imagemagickTest.zip)

**Comment by fmw42** [comment]
In IM 7, use magick composite.  But you should really just use magick in place of convert and use the convert equivalent of the composite.  See https://imagemagick.org/Usage/compose/#compose

Note the first two images are swapped between magick version and the magick convert version.

**Reply by JamesOsborn-SE** [reply]
Thanks for the quick reply 

Good info to know, but this yields the same result. 

For the folks playing at home, the commands are:
```shell
magick D.png L.png rect_mask.png -composite oof.png
magick -debug all D.png L.png rect_mask.png -composite oof-debug.png   
```

**Reply by JamesOsborn-SE** [reply]
I added a zip with the images.

**Reply by JamesOsborn-SE** [reply]
Also, if I create the mask as an 8 bit PNG it seems to work. I'd call this a workaround. It may help in troubleshooting the issue, I hope.

```shell
convert -size ${width}x${height} xc:black -fill white -draw "polygon 0,0 0,${height} ${width},${height}" PNG8:r.png
```
